### PR TITLE
Version 1 Commit

### PR DESCRIPTION
**Pull request by VishalBudhani**
Added the basic functionality for the implementation.

### CODE REVIEW

File: Exercises.Tests/CacheTests.cs [hunk 1]
```diff
@@ -47,7 +47,8 @@ public class CacheTests  {
             var item4 = new ImageDescription() { InstanceUid = Guid.NewGuid().ToString() };
             cache.PutItem(item4.InstanceUid, item4);
 
-            Assert.IsNull(cache.GetItem(item3.InstanceUid), "least recently used item is not removed from cache");
+            //Vishal - changed this
+            Assert.IsNull(cache.GetItem(item1.InstanceUid), "least recently used item is not removed from cache");
```
Comment (prashanthr84) [inline Exercises.Tests/CacheTests.cs:51]: Line numbers 44 and 45 accesses the item1 and item2. But item3 is not accessed after adding. Hence item3 is the least recently used.
Least recently used means the item which is not used recently. It should be item3

File: Exercises/Caching/Cache.cs [hunk 2]
```diff
@@ -19,28 +21,105 @@ public class Cache {
...
 
         public ImageDescription GetItem(string id)  {
             //todo:// if the item present return the item else return null.
+            if (dictionaryDoublyLinkedList.ContainsKey(id))
+                return dictionaryDoublyLinkedList[id].Data.Value;
             return null;
         }
     }
+
+    internal class DoublyLinkedList
```
Comment (prashanthr84) [inline Exercises/Caching/Cache.cs:54]: No need for this class unless its specifically asked not to use the built in type.
LinkedList<T> in .NET library is already a doubly linked list.

File: Exercises/Caching/Cache.cs [hunk 3]
```diff
@@ -19,28 +21,105 @@ public class Cache {
...
 
         private void RemoveLeastRecentlyUsedItem()  {
-            //todo://
+            var deletedNode = doublyLinkedList.DeleteNodeFromEnd();
+            dictionaryDoublyLinkedList.Remove(deletedNode.Data.Key);
         }
 
         public ImageDescription GetItem(string id)  {
             //todo:// if the item present return the item else return null.
+            if (dictionaryDoublyLinkedList.ContainsKey(id))
```
Comment (prashanthr84) [inline Exercises/Caching/Cache.cs:48]: This is the place where fetched item should be moved to head of the linked list:)

File: Exercises.Tests/CacheTests.cs (hunk 1 above)
Comment (VishalBudhani) [reply Exercises.Tests/CacheTests.cs:51]: Agree

**Pull request closed without merging**